hudi-common/src/main/java/org/apache/hudi/common/util/collection/ExternalSpillableMap.java

```diff
@@ -204,19 +205,21 @@ public R get(Object key) {
   public R put(T key, R value) {
     if (this.currentInMemoryMapSize < maxInMemorySizeInBytes || inMemoryMap.containsKey(key)) {
       if (shouldEstimatePayloadSize && estimatedPayloadSize == 0) {
-        // At first, use the sizeEstimate of a record being inserted into the spillable map.
-        // Note, the converter may over estimate the size of a record in the JVM
+        // At first, use the size estimate of a record being inserted into the Spillable map.
+        // Note, the converter may overestimate the size of a record in the JVM.
         this.estimatedPayloadSize = keySizeEstimator.sizeEstimate(key) + valueSizeEstimator.sizeEstimate(value);
-        LOG.info("Estimated Payload size => " + estimatedPayloadSize);
-      } else if (shouldEstimatePayloadSize && inMemoryMap.size() % NUMBER_OF_RECORDS_TO_ESTIMATE_PAYLOAD_SIZE == 0) {
+        LOG.debug("Estimated Payload size => " + estimatedPayloadSize);
```

Comment:
wondering how come we did not hit this issue so far.

Reply:
The map needs to go back to empty after the initial put and within the next threshold put count for the size re-estimation. The MetaIndex code path is triggering this now I guess.